Image_process/processImage.py

```diff
@@ -0,0 +1,136 @@
+        Resize the image to the specified width and height.
+        :param width: Target width of the image.
+        :param height: Target height of the image.
+        :return: Resized Image object.
+        """
+        if width <= 0 or height <= 0:
+            raise ValueError("Width and height must be positive integers.")
+        #if image isn't square raise a warning and resize to square
+        image = Image.open(self.image_path)
+        if image.size[0] != image.size[1]:
+            print("Warning: Image is not square. Resizing to square.")
+
+        return image.resize((width, height))
+    
+    def save_image(self, image, save_path):
+        """
+        Save the image to the specified path.
+        :param image: Image object to save.
+        :param save_path: File path to save the image.
+        """
+        
+        # Ensure the save directory exists
+        os.makedirs(os.path.dirname(save_path), exist_ok=True)
+        # Save the image
+        image.save(save_path)
+
+class GifProcess:
+    def __init__(self, gif_path):
+        """
+        Initialize the GifProcess class with the path to the GIF.
+        :param gif_path: Path to the GIF file.
+        """
+        self.gif_path = gif_path
+
+    def process_gif(self):
+        """
+        Process the GIF to extract all frames as RGBA images.
+        :return: List of Image objects representing each frame.
+        """
+        gif = Image.open(self.gif_path)
+        frames = []  # List to store all processed frames
+        palette = gif.getpalette()  # Get the palette for the GIF if available
+
+        try:
+            while True:
+                # Create a copy of the current frame
+                new_frame = gif.copy()
+
+                # Apply the palette if the frame is in 'P' mode
+                if new_frame.mode == 'P' and palette:
+                    new_frame.putpalette(palette)
+
+                # Convert the frame to RGBA for consistent processing
+                processed_frame = new_frame.convert('RGBA')
+                frames.append(processed_frame)
+
+                # Move to the next frame
+                gif.seek(gif.tell() + 1)
+        except EOFError:
+            # End of the GIF, all frames processed
+            pass
+
+        return frames
+
+    def resize_frames(self, frames, width=128, height=128):
+        """
+        Resize the list of frames to the specified width and height.
+        :param frames: List of Image objects representing each frame.
+        :param width: Target width of the frames.
+        :param height: Target height of the frames.
+        :return: List of resized Image objects.
+        """
+        resized_frames = []
+
+        for frame in frames:
+            # High-quality resizing
+            resized_frame = frame.resize((width, height), Image.Resampling.LANCZOS) 
+            resized_frames.append(resized_frame)
+
+        return resized_frames
+
+    def save_frames(self, frames, save_path):
```

Comment:
what are the use cases for this method? we do not care about the intermediate frames, the goal of this module should be `input image/gif -> output resized image/gif`